### PR TITLE
feat(documentation): add documentation related to aynschronous deployment

### PR DESCRIPTION
**Pull request by kwent**
Related to https://github.com/deis/controller/pull/1269

### CODE REVIEW

**Comment by bacongobbler** [comment]
A good example is https://deis.com/docs/workflow/applications/managing-app-configuration/#custom-health-checks.

**Comment by kwent** [comment]
@bacongobbler Added some documentation. English is not my primary language, so might need some adjustments.

File: src/applications/deploying-apps.md [hunk 1]
```diff
@@ -63,10 +63,35 @@ Setting                                         | Description
 DEIS_DISABLE_CACHE                              | if set, this will disable the [slugbuilder cache][] (default: not set)
 DEIS_DEPLOY_BATCHES                             | the number of pods to bring up and take down sequentially during a scale (default: number of available nodes)
 DEIS_DEPLOY_TIMEOUT                             | deploy timeout in seconds per deploy batch (default: 120)
+DEIS_DEPLOY_NOW                                 | if set to false, this will disable synchronous deployment (default: no set)
```
Comment (vdice) [inline src/applications/deploying-apps.md]: a bit more proper: "(default: not set)"

File: src/applications/deploying-apps.md [hunk 2]
```diff
@@ -63,10 +63,35 @@ Setting                                         | Description
...
 KUBERNETES_POD_TERMINATION_GRACE_PERIOD_SECONDS | how many seconds kubernetes waits for a pod to finish work after a SIGTERM before sending SIGKILL (default: 30)
 
+### Asynchronous Deployment
+
+By default, `git push` and `deis pull` will create a build, a release and proceed to the deployment.
+Starting v2.13 version, it's possible to skip the deployment and trigger the deployment of a release later on.
+
+#### Disabling Synchronous Deployment
+
+There is two way to disable the synchronous deployment:
```
Comment (vdice) [inline src/applications/deploying-apps.md]: more properly "There are two ways to disable..."

File: src/applications/deploying-apps.md [hunk 3]
```diff
@@ -63,10 +63,35 @@ Setting                                         | Description
...
+
+- Using the command line:
+  - The first one which applies for `git push` and `deis pull` is to set `DEIS_DEPLOY_NOW` to `false` in your application settings.
```
Comment (vdice) [inline src/applications/deploying-apps.md]: perhaps illustrate with an example below this, something like:
```
$ deis config:set DEIS_DEPLOY_NOW=false -a <app>
```

File: src/applications/deploying-apps.md [hunk 4]
```diff
@@ -63,10 +63,35 @@ Setting                                         | Description
...
+  - The second one which only applies for `deis pull`, is to pass `--deploy-now false` option to `deis pull` command.
```
Comment (vdice) [inline src/applications/deploying-apps.md]: reads a bit more proper as: "is to pass the `--deploy-now false` option to the `deis pull` command."

Comment (vdice) [inline src/applications/deploying-apps.md]: Also perhaps illustrate with an explicit example below this, something like:
```
$ deis pull deis/example-go -a <app> --deploy-now false
```

File: src/applications/deploying-apps.md [hunk 5]
```diff
@@ -63,10 +63,35 @@ Setting                                         | Description
...
+- Using the API:
+  - Using the [API](https://deis.com/docs/workflow/reference-guide/controller-api/v2.3/#create-application-build) passing `deploy_now: false` parameter.
+
+Note: `DEIS_DEPLOY_NOW` takes precedence over `--deploy-now` option.
+
+#### Deploying A Release Manually
+
+Deploying a release manually is possible:
+
+- Using the command line with `deis deploy <release-version>`.
```
Comment (vdice) [inline src/applications/deploying-apps.md]: Perhaps illustrate:
```
$ deis releases -a <app>
=== <app> Releases
v1	2017-03-31T17:18:43Z	<user> created initial release

$ deis deploy v1 -a <app>
```

File: src/applications/deploying-apps.md [hunk 6]
```diff
@@ -63,10 +63,35 @@ Setting                                         | Description
...
+
+#### Disabling Synchronous Deployment
+
+There is two way to disable the synchronous deployment:
+
+- Using the command line:
+  - The first one which applies for `git push` and `deis pull` is to set `DEIS_DEPLOY_NOW` to `false` in your application settings.
+  - The second one which only applies for `deis pull`, is to pass `--deploy-now false` option to `deis pull` command.
+- Using the API:
+  - Using the [API](https://deis.com/docs/workflow/reference-guide/controller-api/v2.3/#create-application-build) passing `deploy_now: false` parameter.
```
Comment (vdice) [inline src/applications/deploying-apps.md]: a bit more proper: "Using the API and passing the `deploy_now: false` parameter."

File: src/applications/deploying-apps.md [hunk 7]
```diff
@@ -63,10 +63,35 @@ Setting                                         | Description
...
+
+Note: `DEIS_DEPLOY_NOW` takes precedence over `--deploy-now` option.
```
Comment (vdice) [inline src/applications/deploying-apps.md]: a bit more proper: "... takes precedence over the `--deploy-now` option."

**Comment by mboersma** [comment]
These PRs haven't been updated in a few months. @kwent I'm going to close this, but please re-open if you can address the remaining issues, and thanks again!

**Pull request closed without merging**